airstack-modules/modules/airstack/domain-name/domain-name.ts

```diff
@@ -180,279 +148,260 @@ export namespace domain {
       previousResolverId,
       resolverEntity.address,
-      block,
+      airBlock,
       transactionHash,
-      logIndex,
+      logOrCallIndex,
       domain,
-    )
+    );
+    tnx.save();
   }
 
   /**
    * @dev This function tracks a new TTL transaction
-   * @param node domain node
-   * @param newTTL new TTL
-   * @param chainId chain id
-   * @param blockHeight block number
-   * @param blockHash block hash
-   * @param blockTimestamp block timestamp
-   * @param logIndex event log index
+   * @param block ethereum block
    * @param transactionHash transaction hash
+   * @param logOrCallIndex txn log or call index
+   * @param domainId air domain id
+   * @param newTTL new TTL
    * @param tokenAddress contract address of nft token
-   * @param fromOldRegistry specifies if the event is from the old registry
    */
   export function trackDomainNewTTLTransaction(
-    node: string,
-    newTTL: BigInt,
-    chainId: string,
-    blockHeight: BigInt,
-    blockHash: string,
-    blockTimestamp: BigInt,
-    logIndex: BigInt,
+    block: ethereum.Block,
     transactionHash: string,
+    logOrCallIndex: BigInt,
+    domainId: string,
+    newTTL: BigInt,
     tokenAddress: string,
-    fromOldRegistry: boolean,
   ): void {
+    let chainId = getChainId();
     // get block
-    let block = getOrCreateAirBlock(chainId, blockHeight, blockHash, blockTimestamp);
+    let airBlock = getOrCreateAirBlock(chainId, block.number, block.hash.toHexString(), block.timestamp);
+    airBlock.save();
     // get domain
-    let domain = getOrCreateAirDomain(new Domain(node, chainId, block, tokenAddress));
-    let isMigratedMapping = getOrCreateIsMigratedMapping(domain.id, block.id);
-    if (fromOldRegistry && isMigratedMapping.isMigrated == true) {
-      // this domain was migrated from the old registry, so we don't need to hanlde old registry event now
-      return;
-    }
+    let domain = getOrCreateAirDomain(new Domain(domainId, chainId, airBlock, tokenAddress));
     // get previous ttl
     let oldTTL: BigInt | null = null;
-    if (domain.ttl) {
-      oldTTL = domain.ttl;
+    // load extra entity for ttl
+    let extraId = domainId.concat("-").concat(AIR_EXTRA_TTL);
+    let extra = AirExtra.load(extraId);
+    if (extra != null) {
+      // if exists, assign oldTTL and update value with newTTL
+      oldTTL = BigInt.fromString(extra.value);
+      extra.value = newTTL.toString();
+    } else {
+      // else create new extra entity for ttl
+      extra = createAirExtra(AIR_EXTRA_TTL, newTTL.toString(), extraId);
+    }
+    extra.save();
+    // update domain extras
+    let extrasArray = new Array<string>();
+    if (domain.extras == null) {
+      extrasArray.push(extra.id);
+    } else {
+      extrasArray = domain.extras!;
+      extrasArray.push(extra.id);
     }
-    // update domain ttl
-    domain.ttl = newTTL;
-    domain.lastBlock = block.id;
+    domain.extras = extrasArray;
```

Comment:
Evaluate this very carefully, the `extras` could have more that just TTL, we do not want to loose it with new array assignment.
In this case I think its fine, but still I think we should have the complete code that takes care of such scenario.
May be have a util function that can handle this.

Reply:
it could be the case in socials subgraphs, and the module there is handling extras dynamically. Here, in domain-name, the function which uses extras is `trackDomainNewTTLTransaction` and only records ttl data. This will work and is covered in unit tests. Have also handled multiple ttl events for same extras entity in the test case now.

Reply:
done